### ISSUE TITLE
Some offline-cache dependancies are not being tarballed, and are being inserted into the yarn.lock as remote urls

### ISSUE THREAD

**Issue by rjb8682** [yarnpkg/yarn#2690]
**Do you want to request a *feature* or report a *bug*?**

Report a potential bug

**What is the current behavior?**
After walking through the steps from [this guide](https://yarnpkg.com/blog/2016/11/24/offline-mirror/), it seems that some dependancies are being stored into the lock file as the remote urls instead of saving them as tarballs. 

**If the current behavior is a bug, please provide the steps to reproduce.**

```
> yarn config set yarn-offline-mirror ./npm-packages-offline-cache
> mv ~/.yarnrc ./
> rm -rf node_modules/ yarn.lock
> yarn install
```
**What is the expected behavior?**
All dependancies, including the dependancies dependancies, should be stored in tarball form in the specified offline-cache folder.

**Please mention your node.js, yarn and operating system version.**
```
nodejs: 4.5.0
yarn: 0.19.1
OS: MacOS 10.12.3
```

**Comment by rjb8682** [comment]
I noticed this ticket was made a while ago but was closed. Seems fairly similar. #318

**Comment by bestander** [comment]
I think there is still a problem with caches.
Try `yarn cache clean`, remove yarn.lock and try again

**Comment by rjb8682** [comment]
That did it! I appreciate the help. Hopefully this can get fixed up in the future, but at least now i have a solution if it sticks.

Thanks a ton

**Comment by elliotchi** [comment]
+1

**Comment by bestander** [comment]
I think this happens when you switch between a repo with offline cache and the one without it.
The non offline cache repo creates a global cache folder for every package in a non offline mirror compatible way.
We'll need to sort it out the sooner the better

**Comment by bestander** [comment]
The more reasons to keep yarn.lock unchanged when using offline mirror and making substitutions based on current config during install time.

**Comment by skevy** [comment]
@bestander what do you see as the next steps to resolve this issue? This is plaguing us pretty bad at Exponent -- happens every single day to some degree.

**Comment by bestander** [comment]
This is on that top of our list.
@arcanis, new core Yarn team member is working on an RFC and a fix https://github.com/yarnpkg/rfcs/pull/51.

**Comment by bestander** [comment]
This is fixed in 0.23.
Now yarn.lock contains remote URL always and the tarballs should be saved correctly
